domain-models-runtime/src/main/java/org/folio/rest/persist/cache/CachedConnectionManager.java

```diff
@@ -0,0 +1,111 @@
+package org.folio.rest.persist.cache;
+
+import io.vertx.core.Vertx;
+import io.vertx.pgclient.PgConnection;
+import io.vertx.sqlclient.Pool;
+import io.vertx.core.Future;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.folio.rest.persist.PostgresClient;
+import org.folio.rest.tools.utils.Envs;
+
+import java.util.Optional;
+
+public class CachedConnectionManager {
+  private static final Logger LOG = LogManager.getLogger(CachedConnectionManager.class);
+  private final ConnectionCache connectionCache = new ConnectionCache();
+  private final int maxPoolSize;
+  private int connectionReleaseDelaySeconds = -1;
+
+  public CachedConnectionManager() {
+    this.maxPoolSize = getMaxSharedPoolSize();
+  }
+
+  public int getCacheSize() {
+    return connectionCache.size();
+  }
+
+  public void clearCache() {
+    connectionCache.clear();
+    LOG.debug("Cleared connection manager");
+  }
+
+  public void removeFromCache(CachedPgConnection connection)  {
+    connectionCache.remove(connection);
+  }
+
+  public void tryAddToCache(CachedPgConnection connection) {
+    connectionCache.tryAdd(connection);
+    this.tryRemoveOldestAvailableConnectionAndClose();
+  }
+
+  public Future<PgConnection> getConnection(Vertx vertx, Pool pool, String schemaName, String tenantId) {
+    if (! PostgresClient.isSharedPool()) {
+      LOG.debug("Not in shared pool mode");
+      return pool.getConnection().map(PgConnection.class::cast);
+    }
+
+    if (this.connectionReleaseDelaySeconds == -1) {
```

Comment:
is there a way to ensure this is set before calling `getConnection`?

Reply:
It will be set before since it is set in `PostgresClient.createPgPool`[here](https://github.com/folio-org/raml-module-builder/pull/1142/files#diff-0de14d572773e0367110bc28030981f9105376ca37b0b92716f5389f819cf8c6R627) but I think you are right it is a code smell.

I have 2 ways of dealing with configuration/env values. The way shown here, and the way I get the `maxSharedPoolSize` in the constructor you rightly take issue with below. There shouldn't be 2 ways.

I think I can statically define both of these values in the class where they either read from env or the default if not present.